packages/mobile/src/identity/feelessVerification.ts

```diff
@@ -1051,25 +1088,26 @@ export function* feelessCompleteAttestation(
 
   Logger.debug(TAG + '@feelessCompleteAttestation', `Completing code for issuer: ${code.issuer}`)
 
-  // TODO: Refactor this to check for state change in a smart way
-  // Needed to ensure that codes inputted in quick succession are not
-  // assigned the same nonce by Komenci
+  // Make each concurrent completion attempt wait a sec for where they are relative to other codes
+  // to ensure `processingInputCode` has enough time to properly gate the tx. 0-index code
+  // will have 0 delay, 1-index code will have 1 sec delay, etc.
+  yield delay(codePosition * 1000)
   while (yield select(feelessProcessingInputCodeSelector)) {
-    yield delay(100)
+    yield delay(Math.random() * 1000)
   }
 
   yield put(feelessProcessingInputCode(true))
   const completeTxResult: Result<TransactionReceipt, FetchError | TxError> = yield call(
-    [komenciKit, komenciKit.completeAttestation],
+    submitCompleteTxAndRetryOnRevert,
+    komenciKit,
     mtwAddress,
-    phoneHashDetails.phoneHash,
-    code.issuer,
-    code.code
+    phoneHashDetails,
+    code
   )
+  yield put(feelessProcessingInputCode(false))
```

Comment:
I think we probably just want a good mutex solution in a few places but this will work